lib/Dialect/Wave/IR/WaveAttrs.cpp

```diff
@@ -163,6 +164,105 @@ bool WaveHyperparameterAttr::hasSymbol(StringRef symbolName) const {
   return getMapping().get(symbolName) != nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// WaveDistributedShapeAttr
+//===----------------------------------------------------------------------===//
+
+std::optional<llvm::SmallVector<int64_t>>
+wave::DistributedShapeAttr::getResolvedShape(
+    wave::WaveHyperparameterAttr hyper) const {
+  auto values = wave::resolveSymbolNames(getSymbolNames(), hyper);
+  return wave::evaluateMapWithSymbols(getShape(), *values);
```

Comment:
```suggestion
  if (!values)
    return std::nullopt;
  return wave::evaluateMapWithSymbols(getShape(), *values);
```

there's no guarantee the previous function succeeds.